inginious/agent/docker_agent/__init__.py

```diff
@@ -524,17 +527,19 @@ def __new_job_sync(self, message, future_results):
                 self._create_safe_task(close_and_delete(student_container_id_loop))
             del self._student_containers_for_job[message.job_id]
 
-            # Allow other container to reuse the ssh port this container has finished to use
-            if container_id in self._running_ssh_debug:
-                self._ssh_ports.add(self._running_ssh_debug[container_id])
-                del self._running_ssh_debug[container_id]
+            # Allow other container to reuse the external ports this container has finished to use
+            if container_id in self._assigned_external_ports:
+                for p in self._assigned_external_ports[container_id]:
+                    self._external_ports.add(p)
+                del self._assigned_external_ports[container_id]
 
             # Verify if the container was killed, either by the client, by an OOM or by a timeout
             killed = await self._timeout_watcher.was_killed(container_id)
             if container_id in self._containers_killed:
                 killed = self._containers_killed[container_id]
-                if container_id in self._running_ssh_debug:
-                    self._ssh_ports.add(self._running_ssh_debug[container_id])
+                if container_id in self._assigned_external_ports:
```

Comment:
I think this is dead code and can be safely removed. This is already done at line 530.